src/integration/test_suites/smoke_test_fc_secret_partition_zeroization/smoke_test_fc_secret_partition_zeroization.c

```diff
@@ -0,0 +1,168 @@
+
+  const uint32_t data[4] = {0xdeadbeef, 0xcafebabe, 0x12345678, 0xabababab};
+  uint32_t read_data[4];
+  uint32_t zero_r_data[2];
+  uint32_t zeroize_num_times = 3;
+
+  // Step 1. Write a full transition token into the partition.
+  dai_wr(fuse_address, data[0], data[1], 64, 0);
+  dai_wr(fuse_address+8, data[2], data[3], 64, 0);
+
+  // Step 2. Read back the token and verify it is equal to the token written in Step 1.
+  //         This works as long as the secret partition has not been locked.
+  dai_rd(fuse_address, &read_data[0], &read_data[1], 64, 0);
+  dai_rd(fuse_address+8, &read_data[2], &read_data[3], 64, 0);
+  for (uint32_t idx = 0; idx < 4; idx++) {
+    if (data[idx] != read_data[idx]) {
+      VPRINTF(LOW, "ERROR: incorrect fuse data[index=%0d]: expected: %08X actual: %08X\n", idx, data[idx], read_data[idx]);
+      goto epilogue;
+    }
+  }
+  // Step 3. Verify that the partition's digest register is 0.
+  uint32_t digest[2];
+  digest[0] = lsu_read_32(OTP_CTRL_SECRET_PROD_PARTITION_0_DIGEST_DIGEST_0);
+  digest[1] = lsu_read_32(OTP_CTRL_SECRET_PROD_PARTITION_0_DIGEST_DIGEST_1);
+  if (digest[0] != 0 || digest[1] != 0) {
+    VPRINTF(LOW, "ERROR: digest is not 0\n");
+    goto epilogue;
+  }
+
+  // Step 4. Lock the partition by calculating a hardware digest.
+  calculate_digest(base_address, 0);
+
+  // Step 5. Reset the RTL.
+  reset_fc_lcc_rtl();
+  wait_dai_op_idle(0);
+
+  // Step 6. Try to read the token and verify that it results in an error as the
+  //         partition is secret and locked, thus not accessible by software anymore.
+  dai_rd(fuse_address, &read_data[0], &read_data[1], 64, OTP_CTRL_STATUS_DAI_ERROR_MASK);
+
+  // Step 7. Try to write a value into the partition and verify that it results in an error
+  //         as the partition has been locked in Step 5.
+  dai_wr(fuse_address, data[0], data[1], 64, OTP_CTRL_STATUS_DAI_ERROR_MASK);
+
+  // Step 8. Read back the digest from the partition's digest register and verify it
+  //         is non-zero.
+  digest[0] = lsu_read_32(OTP_CTRL_SECRET_PROD_PARTITION_0_DIGEST_DIGEST_0);
+  digest[1] = lsu_read_32(OTP_CTRL_SECRET_PROD_PARTITION_0_DIGEST_DIGEST_1);
+  if (digest[0] == 0 || digest[1] == 0) {
+    VPRINTF(LOW, "ERROR: digest is 0\n");
+    goto epilogue;
+  }
+
+  // Step 9. Apply zeroization multiple_times
+  for(uint32_t num_times = 0; num_times < zeroize_num_times; num_times++) {
+    // Zeroize fuse.
+    for (uint32_t addr = partitions[VENDOR_SECRET_PROD_PARTITION].address; addr < partitions[VENDOR_SECRET_PROD_PARTITION].digest_address; addr += (partitions[VENDOR_SECRET_PROD_PARTITION].granularity / 8)) {
+      dai_zer(addr, &read_data[0], &read_data[1], partitions[VENDOR_SECRET_PROD_PARTITION].granularity, 0);
+      if (read_data[0] != 0xFFFFFFFF || (partitions[VENDOR_SECRET_PROD_PARTITION].granularity> 32 && read_data[1] != 0xFFFFFFFF)) {
+        VPRINTF(LOW, "ERROR: [partitions[SECRET_LC_TRANSITION_PARTITION] ]fuse is not zeroized\n");
```

Comment:
This seems to be zeroizing the wrong partition. Also, the code can be simplified as the granularity of the partition is fixed.